libsolidity/codegen/ArrayUtils.cpp

```diff
@@ -335,9 +335,14 @@ void ArrayUtils::copyArrayToMemory(ArrayType const& _sourceType, bool _padToWord
 		if (baseSize > 1)
```

Comment:
The encoding in calldata is different for arrays than in memory. A value of type `uint[][]` is encoded in memory as follows: length followed by a list of absolute memory addresses of the elements. In calldata, it is a list of offsets from the current "block".